starknet_py/net/account/account.py

```diff
@@ -0,0 +1,394 @@
+        """
+        constructor_calldata = constructor_calldata or []
+
+        deploy_account_tx = DeployAccount(
+            class_hash=class_hash,
+            contract_address_salt=contract_address_salt,
+            constructor_calldata=constructor_calldata,
+            version=self._version,
+            max_fee=0,
+            signature=[],
+            nonce=0,
+        )
+
+        max_fee = await self._get_max_fee(
+            transaction=deploy_account_tx, max_fee=max_fee, auto_estimate=auto_estimate
+        )
+        deploy_account_tx = dataclasses.replace(deploy_account_tx, max_fee=max_fee)
+        signature = self.signer.sign_transaction(deploy_account_tx)
+        return _add_signature_to_transaction(deploy_account_tx, signature)
+
+    async def execute(
+        self,
+        calls: Calls,
+        *,
+        max_fee: Optional[int] = None,
+        auto_estimate: bool = False,
+    ) -> SentTransactionResponse:
+        """
+        Takes calls and executes transaction
+
+        :param calls: Single call or list of calls
+        :param max_fee: Max amount of Wei to be paid when executing transaction
+        :param auto_estimate: Use automatic fee estimation, not recommend as it may lead to high costs
+        :return: SentTransactionResponse
+        """
+        execute_transaction = await self.sign_invoke_transaction(
+            calls, max_fee=max_fee, auto_estimate=auto_estimate
+        )
+        return await self._client.send_transaction(execute_transaction)
+
+    def sign_message(self, typed_data: TypedData) -> List[int]:
+        """
+        Sign an TypedData TypedDict for off-chain usage with the starknet private key and return the signature
+        This adds a message prefix, so it can't be interchanged with transactions
+
+        :param typed_data: TypedData TypedDict to be signed
+        :return: The signature of the TypedData TypedDict
+        """
+        return self.signer.sign_message(typed_data, self.address)
+
+    async def verify_message(self, typed_data: TypedData, signature: List[int]) -> bool:
+        """
+        Verify a signature of a TypedData TypedDict
+
+        :param typed_data: TypedData TypedDict to be verified
+        :param signature: signature of the TypedData TypedDict
+        :return: true if the signature is valid, false otherwise
+        """
+        typed_data_dataclass = TypedDataDataclass.from_dict(typed_data)
+        message_hash = typed_data_dataclass.message_hash(account_address=self.address)
+        return await self._verify_message_hash(message_hash, signature)
+
+
+SignableTransaction = TypeVar(
+    "SignableTransaction", InvokeFunction, Declare, DeployAccount
+)
+
+
+def _add_signature_to_transaction(
+    tx: SignableTransaction, signature: List[int]
+) -> SignableTransaction:
+    return dataclasses.replace(tx, signature=signature)
+
+
+def _merge_calls(calls: Iterable[Call]) -> List:
+    def parse_call(
```

Comment:
It seems that this function doesn't have to be nested here, it doesn't use any variables form the clojure.

Reply:
I guess it was just nested because it's only used in this method, but we can changed it.